lib/src/injector_base.dart

```diff
@@ -132,6 +142,39 @@ class Injector {
     return instance;
   }
 
+  ///
+  /// This method checks if there is a registered dependency
+  ///
+  bool exists<T>({String dependencyName = ""}) {
+    String dependencyKey = _getIdentity<T>(dependencyName);
+
+    return _factoryMap.containsKey(dependencyKey);
+  }
+
+  ///
+  /// This method removes one dependency by the key
+  ///
+  void clearByKey<T>({String dependencyName = ""}) {
```

Comment:
We are not using the term "key" anywhere, how about calling this one just `clear` or `remove`